### PR TITLE
[libuuid] Cross support for Conan v1 and v2

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **libuuid/1.0.3**

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/libuuid/all/conanfile.py [hunk 1]
```diff
@@ -15,66 +18,66 @@ class LibuuidConan(ConanFile):
...
-        files.apply_conandata_patches(self)
-        with files.chdir(self, self._source_subfolder):
-            self.run("{} -fiv".format(tools.get_env("AUTORECONF")), run_environment=True)
-        autotools = self._configure_autotools()
-        autotools.make()
+        apply_conandata_patches(self)
+        autotools = Autotools(self)
+        autotools.autoreconf()
+        autotools.configure()
+        autotools.make
```
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/libuuid/all/conanfile.py:71]: Yes, but, still it's duplicated with https://github.com/conan-io/conan-center-index/pull/13130 I gonna close this PR.

File: recipes/libuuid/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -1,9 +1,8 @@
 cmake_minimum_required(VERSION 3.1)
```
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/libuuid/all/test_package/CMakeLists.txt:1]: Good catch, 3.8 is the correct one.

**Comment by uilianries** [comment]
Duplicated with https://github.com/conan-io/conan-center-index/pull/13130

**Comment by michalwidera** [comment]
dup has commited. Can we resolve this?

**Pull request closed without merging**